### ISSUE TITLE
Extend PeerBook with Reputation

### ISSUE THREAD

**Issue by wanderer** [wanderer/js-libp2p-gossip-discovery#2]
It would be nice to have an instance of [PeerBook](https://github.com/libp2p/js-peer-book) that also tracked the reputation of peers. It could methods such as `updateReputation(id, int)`  and `getReputation(id)` and `setReputationThreshold(int)` which would set a minimal reputation, if a peer drops below this rep, then it would get evected from the peerBook

**Comment by sdtsui** [comment]
Thanks @wanderer - will see about grabbing this after #3. 

Notes to self: 
- for now, makes most sense to subtract reputation in `_peerDiscovery`: [L87](https://github.com/wanderer/js-libp2p-gossip-discovery/blob/master/index.js#L87)
- May want to use some sort of log scale for reputation.
